src/features/codiga-file-suggestion.ts

```diff
@@ -0,0 +1,141 @@
+import * as vscode from "vscode";
+import {
+  CODIGA_RULES_FILE,
+  DEFAULT_JAVASCRIPT_RULESET_CONFIG,
+  DEFAULT_PYTHON_RULESET_CONFIG,
+  INFO_MESSAGE_CODIGA_FILE,
+  INFO_MESSAGE_CODIGA_FILE_ACTION_CREATE,
+  INFO_MESSAGE_CODIGA_FILE_ACTION_IGNORE,
+  INFO_MESSAGE_CODIGA_FILE_KEY,
+  ROSIE_SUPPORTED_LANGUAGES,
+  VALUE_STRING_TRUE,
+} from "../constants";
+import { addCreateCodigaYamlRecord } from "../graphql-api/add-create-codiga-yaml-record";
+import { Language } from "../graphql-api/types";
+import { getWorkspaceRosieLanguage } from "../rosie/rosieUtils";
+import getFileUri, { doesFileExist } from "../utils/fileUtils";
+import { getFromLocalStorage, setToLocalStorage } from "../utils/localStorage";
+
+/**
+ * check whether the user has ignored codiga file suggestion for their workspace
+ * @returns boolean
+ */
+export function hasUserIgnoredCodigaSuggestion(): boolean {
+  return (
+    getFromLocalStorage(INFO_MESSAGE_CODIGA_FILE_KEY) === VALUE_STRING_TRUE
+  );
+}
```

Comment:
is `getFromLocalStorage` for workspace only?

Reply:
Yes.

localStorage can be at a workspace or global level.

[The only localStorage we use was set here in `extension.ts`](https://github.com/codiga/vscode-plugin/blob/4d3db643320a61039317958449df7cc6719ff9a2/src/extension.ts#L56)

If needed, we can add and maintain a global version too.